frontend/src/metabase/visualizations/lib/settings/visualization.js

```diff
@@ -52,10 +53,12 @@ function getSettingDefintionsForSeries(series) {
     ...COMMON_SETTINGS,
     ...(visualization.settings || {}),
   };
-  for (const id in definitions) {
-    definitions[id].id = id;
-  }
-  return definitions;
+
+  return produce(definitions, draftDefinitions => {
```

Comment:
I had to change this because [immer autofreezes objects](https://immerjs.github.io/immer/freezing/, and because of my use of produce in `ObjectDetail.jsx` it caused problems here

Reply:
Do you still need to do this?